internal/provider/provider.go

```diff
@@ -291,11 +298,14 @@ func providerConfigure(p *schema.Provider) schema.ConfigureContextFunc {
 
 			// Doc Links
 			ClientSecretDocsLink: "https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_client_secret",
+
+			// MSAL opt-in
+			UseMicrosoftGraph: d.Get("use_msal").(bool),
```

Comment:
fwiw this could become:

```suggestion
			UseMicrosoftGraph: func() bool {
				if features.ThreePointOh() {
					return true
				}
				
				return d.Get("use_msal").(bool)
			}(),
```

Reply:
Maybe this would be better as a DefaultFunc in the schema?

Reply:
Sure that'd work 👍